mu_repo/action_clone.py

```diff
@@ -0,0 +1,127 @@
+from mu_repo.execute_command import ExecuteCommand
+import os
+
+#===================================================================================================
+# Run
+#===================================================================================================
+def Run(params):
+    '''
+    mu-repo can deal with cloning a repository and other referenced repositories recursively.
+
+    To work this way, users are expected to commit the .mu_repo files they rely on and set
+    the base urls that users work with.
+
+    i.e.:
+
+    Say that someone is working with 3 projects: A, B and C and C depends on B and B depends on A
+    and they're all available on https://github.com/myuser
+
+    The urls in this case to be checked out would be something as:
+
+    ssh://git@github.com:myuser/projectA.git
+
+    https://github.com/myuser/projectA.git
+
+    So, the url: https://github.com/myuser or git@github.com:myuser has to be configured as the
+    'base' url for clones.
+
+    mu-repo in this case will work with git to provide that configuration, so, one should first
+    add that remote for mu-repo to work with:
+
+    git config --global --add mu-repo.remote-host ssh://git@github.com:myuser
+
+    Note that it's possible to add as many urls as wanted (they'll all be checked later on
+    when cloning the project). Note that multiple remote hosts may be added that way.
+
+    To check what are the actual urls that mu-repo will use (and the order in which they'll be
+    tried, one can do):
+
+    git config --get-regexp mu-repo.remote-host
+    '''
+
+    args = params.args
+    assert args[0] == 'clone'
+    args = args[1:]
+
+    repos = []
+    other_cmd_line_args = []
+
+    for arg in args:
+        if not arg.startswith('-') and not '@' in arg and not ':' in arg and not '/' in arg:
+            repos.append(arg)
+        else:
+            other_cmd_line_args.append(arg)
+
+
+    git = params.config.git
+
+    if not repos:
+        # We do not have any base: do a regular clone
+        ExecuteCommand([git] + params.args)
+        return
+
+    remote_hosts = params.config.remote_hosts
+
+    if not remote_hosts:
```

Comment:
Hmmm not sure if it wouldn't be more helpful instead to display a message saying that no mu-repo.remote-host is configured, and even providing an example on how to do so.

But I also understand that in general the approach is to just forward subcommands to git... this makes me wonder if perhaps this should be another command, mclone perhaps?